client/internal/packagessyncer.go

```diff
@@ -280,6 +280,12 @@ func (s *packagesSyncer) downloadFile(ctx context.Context, pkgName string, file
 		return fmt.Errorf("cannot download file from %s: %v", file.DownloadUrl, err)
 	}
 
+	if file.Headers != nil && len(file.Headers.Headers) > 0 {
```

Comment:
Nit, I think the second check is unnecessary?
```suggestion
	if file.Headers != nil {
```

Reply:
Yeah, you're right, removed it on d88b449cb455b8541a4c17305e24e61515dbf7a7